index.js

```diff
@@ -1,21 +1,35 @@
-// Inspired by (but complete rewrite of) requirejs-metagen
-let _ = require('lodash/fp')
-let Promise = require('bluebird')
-let readDir = Promise.promisify(require('recursive-readdir'))
-let fs = Promise.promisifyAll(require('fs'))
-
-// Path Utils
-let filesRelative = (dir, ex) => readDir(dir, ex).map(x => x.slice(dir.length))
+// Inspired by (but complete rewrite of) requirejs-metagen.
+let _       = require('lodash/fp') // Add lodash helper functions.
+let Promise = require('bluebird')  // Replace promises with the Bluebird Promises library. ("Bluebird cuts corners" -- yikes!)
+let readDir = Promise.promisify(require('recursive-readdir')) // Lists all files in a certain directory and all its sub-directories.
+let fs      = Promise.promisifyAll(require('fs')) // Add the filesystem library.
```

Comment:
I don't think there's value in leaving comments for descriptions of libraries we use. If you're not sure what they are you should look them up. A dev working on this project should not need a comment to know what `lodash/fp` or `bluebird` are.